app/Services/PgBackRestService.php

```
@@ -0,0 +1,474 @@
+            return 'full';
+        }
+
+        // Find last full backup
+        $lastFull = collect($backups)->where('type', 'full')->sortByDesc('timestamp')->first();
+
+        if (! $lastFull) {
+            return 'full';
+        }
+
+        // If last full backup is older than 7 days, do a full backup
+        $lastFullTime = \Carbon\Carbon::parse($lastFull['timestamp']['start']);
```

PHPMD (2.15.0)

Avoid using static access to class '\Carbon\Carbon' in method 'determineBackupType'. (undefined)

(StaticAccess)